### ISSUE TITLE
TenantTestCase does not call setUpTestData

### ISSUE THREAD

**Issue by taobojlen** [django-tenants/django-tenants#813]
We use [`setUpTestData` ](https://docs.djangoproject.com/en/4.1/topics/testing/tools/#django.test.TestCase.setUpTestData) to initialize data in our tests. It's faster than `setUp` since it's only run once per test case, rather than once per test.

In Django, `setUpTestData` is called in `setUpClass`: https://github.com/django/django/blob/main/django/test/testcases.py#L1448

However, when using django-tenants' `TenantTestCase`, `setUpClass` does *not* call `setUpTestData`.

I can think of two ways to fix this:

1. Call `super().setUpClass()` in `TenantTestCase`
2. Call `setUpTestData()` in `TenantTestCase`

I'm not sure if option 1 will work, but if it does, it would ensure that `TenantTestCase` continues to behave like Django's `TestCase`.

**Comment by Brechard** [comment]
yes, this is an issue, I managed to solve it thanks to a combination of several responses in the issues, I can find the main one to link that suggested calling super().setUpClass(). What I did was creating a TestBase class that I reuse everywhere and has the following code (I use subdomain folders so there is some extra code that I have deleted here, if you need it I can provide it too). Hope this helps you, I can also help if contribution is allowed:

```
from django.test import TestCase
from django_tenants.test.cases import FastTenantTestCase

class TestBase(FastTenantTestCase):
    fixtures = ["fixtures_for_tests.json"]
    tenant_client = None

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

    @classmethod
    def tearDownClass(cls):
        TestCase.tearDownClass()
        super().tearDownClass()

    @classmethod
    def setUp(cls):
        connection.set_tenant(cls.tenant)

    @classmethod
    def setUpClass(cls):
        super().setUpClass()
        connection.set_schema_to_public()
        # Ensure that fixtures are loaded here
        TestCase.fixtures = cls.fixtures
        TestCase.setUpClass()
        connection.set_tenant(cls.tenant)
        cls.setUpTestData()

    @classmethod
    def setUpTestData(cls):
        connection.set_tenant(cls.tenant)
```
